pkg/controller/volume/attachdetach/metrics/metrics.go

```diff
@@ -127,8 +164,28 @@ func (collector *volumeInUseCollector) getVolumeInUseCount() nodeVolumeCount {
 			if err != nil {
 				continue
 			}
-			nodeVolumeMap.add(nodeName, volumePlugin.GetPluginName())
+			nodeVolumeMap.add(pod.Spec.NodeName, volumePlugin.GetPluginName())
 		}
 	}
 	return nodeVolumeMap
 }
+
+func (collector *volumesCollector) getTotalVolumesCount() volumeCount {
+	stateVolumeMap := make(volumeCount)
+	for _, v := range collector.dsw.GetVolumesToAttach() {
+		if plugin, err := collector.volumePluginMgr.FindPluginBySpec(v.VolumeSpec); err == nil {
+			stateVolumeMap.add("desired_state_of_world", plugin.GetPluginName())
+		}
+	}
+	for _, v := range collector.asw.GetAttachedVolumes() {
+		if plugin, err := collector.volumePluginMgr.FindPluginBySpec(v.VolumeSpec); err == nil {
+			stateVolumeMap.add("actual_state_of_world", plugin.GetPluginName())
+		}
```

Comment:
It is probably still worth adding a check for `plugin != nil` here I think. I think there are some corner cases where plugin could be `nil` and yet error is nil too.

Reply:
Done.